packages/@tinacms/toolkit/src/packages/core/media-store.default.ts

```diff
@@ -42,3 +54,229 @@ export class DummyMediaStore implements MediaStore {
     // Unnecessary
   }
 }
+
+export class TinaMediaStore implements MediaStore {
+  fetchFunction = (input: RequestInfo, init?: RequestInit) => {
+    return fetch(input, init)
+  }
+
+  private api: any
+  private cms: CMS
+  private isLocal: boolean
+  private url: string
+  constructor(cms: CMS) {
+    this.cms = cms
+  }
+
+  setup() {
+    if (!this.api) {
+      this.api = this.cms?.api?.tina
+
+      this.isLocal = !!this.api.isLocalMode
+
+      const contentApiUrl = new URL(this.api.contentApiUrl)
+      this.url = `${contentApiUrl.origin}/media`
+      if (!this.isLocal) {
+        if (this.api.options?.tinaioConfig?.assetsApiUrlOverride) {
+          const url = new URL(this.api.assetsApiUrl)
+          this.url = `${url.origin}/v1/${this.api.clientId}`
+        }
+      }
+    }
+  }
+
+  async isAuthenticated() {
+    this.setup()
+    return await this.api.isAuthenticated()
+  }
+
+  accept = 'text/*,  application/*, image/*'
```

Comment:
Is this accurate?  I seem to recall we were limiting the filetypes to just image formats.